### PR TITLE
Feature: create Connection from authsource

### PR DESCRIPTION
**Pull request by tvdijen**
Solves #36

### CODE REVIEW

**Comment by tvdijen** [comment]
@thijskh You recently had a similar case where your touched one thing and then a totally unrelated unit test failed... I just can't remember how it was solved.. Can you?

File: src/Auth/Source/Ldap.php [hunk 1]
```diff
@@ -58,7 +58,7 @@ public function __construct(array $info, array $config)
             'authsources[' . var_export($this->authId, true) . ']'
         );
 
-        $this->connector = $this->resolveConnector();
+        $this->connector = ConnectorFactory::fromAuthSource($this->authId);
```
Comment (tvdijen) [inline src/Auth/Source/Ldap.php:61]: Changes to this file is what's causing the LdapTest to fail...

**Comment by tvdijen** [comment]
@thijskh Can I lend your eyes here? I think the LdapTest was broken before this change..
I think you had a similar case recently, but I can't remember how we fixed it

**Comment by thijskh** [comment]
Well, it passed before fixing the Psalm issues so it should be possible to determine from that what failed? Or even apply them partially until it fails..

**Comment by tvdijen** [comment]
It's this line;
https://github.com/simplesamlphp/simplesamlphp-module-ldap/pull/37/files#diff-550049c1a863aae4e69b318b9a173615e5e67d2fa6766c4beb77a50400ebdab9R61

I wonder if the test was actually run before, because when I set an authsource config in the test, it fails

**Comment by tvdijen** [comment]
Nailed it!

**Comment by tvdijen** [comment]
One thing that might be debatable is that I've added getAttributes on the AuthSource instead of somewhere else.. I wasn't sure what the best place is.

**Review by thijskh — APPROVED** [review]

**Pull request merged**